src/Resource/FilePlugin.php

```diff
@@ -56,6 +56,7 @@ public function populate(Source $source, Template $_template = null) {
 	 * @param Source $source source object
 	 */
 	public function populateTimestamp(Source $source) {
+		$path = false;
```

Comment:
How about this?
```
public function populateTimestamp(Source $source) {
		$path = $this->getFilePath($source->name, $source->getSmarty(), $source->isConfig);

		if (!$source->exists) {
			$source->timestamp = $source->exists = ($path !== false && is_file($path));
		}
		if ($source->exists) {
			$source->timestamp = ($path !== false && filemtime($path));
		}
	}
```

Reply:
Yes, i see,
Thinking of timestamp is only int, would not reset it to (bool(true)==1) 
'($path !== false && filemtime($path)' is always bool, changed it to an int value.
And explicit reset to 0 (or maybe is null better) if absent

```
	public function populateTimestamp(Source $source) {
              $path = $this->getFilePath($source->name, $source->getSmarty(), $source->isConfig);
              if (!$source->exists) {
                  $source->exists = ($path !== false && is_file($path));
              }
              if ($source->exists && $path !== false) {
                  $source->timestamp = filemtime($path);
              } else {
                  $source->timestamp = 0;
              }
	}
```

Reply:
Yes, you are right. Can you add this to your PR?

Reply:
done